"""
Задание 2.

Создайте собственный!!! класс-исключение, обрабатывающий ситуацию деления на нуль.
Проверьте его работу на данных, вводимых пользователем. При вводе пользователем нуля
в качестве делителя программа должна корректно обработать эту ситуацию и не завершиться с ошибкой.
"""
class DivError(Exception):
    def __init__(self, txt):
        self.txt = txt

num1 = int(input('Введите первое число: '))
num2 = int(input('Введите второе число: '))
try:
    if num2 == 0:
        raise DivError('На ноль делить нельзя!')
except DivError as err:
    print(err)
else:
    print(f'Деление чисел равно: {num1 / num2}')